src/custom-elements/src/components/phone-input/phone-input.tsx

```diff
@@ -0,0 +1,286 @@
+        country => country.countryAbbr === this.defaultCountry
+      );
+      this._prefix = `+${this._country.dialCode}`;
+
+      if (this.value) {
+        this._suffix = new AsYouType(this._country.countryAbbr).input(
+          this.value.substring(this.value.indexOf('-') + 1)
+        );
+
+        const valuePrefix = this.value.substring(0, this.value.indexOf('-'));
+
+        if (valuePrefix !== this._prefix) {
+          throw new Error(
+            `Country code prefix of the provided value (${valuePrefix}) does not match the provided default country's prefix (${this._prefix}).`
+          );
+        }
+      }
+    } else {
+      throw new Error(
+        `${this.defaultCountry} is not a valid country for phone input.`
+      );
+    }
+  }
+
+  _suffixInputChangeHandler = (event: Event): void => {
+    event.stopPropagation();
+    this._suffix = (event.target as HTMLInputElement).value;
+    if (!isValidPhoneNumber(this._suffix, this._country.countryAbbr)) {
+      this.chiNumberInvalid.emit();
+    }
+    this.value = this._getValue();
+    this.chiChange.emit(this.value);
+  };
+
+  _inputHandler = (event: Event): void => {
+    event.stopPropagation();
+    this._suffix = new AsYouType(this._country.countryAbbr).input(
+      (event.target as HTMLInputElement).value
+    );
+    this.value = this._getValue();
+    this.chiInput.emit(this.value);
+  };
+
+  _prefixChangeHandler(country: Country): void {
+    this._prefix = `+${country.dialCode}`;
+    this._country = country;
+    this.value = this._getValue();
+    this._isDropdownActive = false;
+    this.chiChange.emit(this.value);
+  }
+
+  _dropdownTriggerHandler(): void {
+    this._isDropdownActive = !this._isDropdownActive;
+    if (this._isDropdownActive) {
+      setTimeout(() => {
+        (this.el.querySelector(
+          'input[type=search]'
+        ) as HTMLInputElement).focus();
+      }, 50);
+    }
+  }
+
+  _closeDropdown = (): void => {
+    this._clickedOnComponent
+      ? (this._clickedOnComponent = false)
+      : (this._isDropdownActive = false);
+  };
+
+  _getValue(): string {
+    return `${this._prefix}-${this._suffix.replace(/[- )(]/g, '')}`;
+  }
+
+  _renderDropdown(): JSX.Element {
+    const searchInput = (
+      <chi-search-input
+        size="sm"
+        placeholder="Search"
+        value={this._search}
+        onChiInput={e => (this._search = (e.target as HTMLInputElement).value)}
+        onChiClean={() => (this._search = '')}
+      ></chi-search-input>
+    );
+    const filteredCountries = this._countries.filter(country => {
+      return (
+        country.name.toLowerCase().indexOf(this._search.toLowerCase()) > -1 ||
+        country.dialCode.indexOf(this._search) > -1 ||
+        ('+' + country.dialCode).indexOf(this._search) > -1
+      );
+    });
+    const countries = (
+      <div class={`${DROPDOWN_CLASSES.MENU_CONTENT}`}>
+        {filteredCountries.map(country => (
+          <a
+            href="javascript:void(0);"
```

Comment:
```suggestion
            href="#"
```
We will also need to add `e.stopPropagation()` or `e.preventDefault()` inside `this._prefixChangeHandler`

<img width="358" alt="Screenshot 2021-11-05 at 17 51 46" src="https://user-images.githubusercontent.com/25136131/140548323-ccc918e2-6ec1-4fcf-8652-8a2d536a4d68.png">